SYCL/Matrix/element_wise_int8_sumLocalRow.cpp

```diff
@@ -0,0 +1,177 @@
+
+              x x x x  x x x x    x x x x  x x x x  x x x x  x x x x    x x x x  x x x x
+              x x x x  x x x x    x x x x  x x x x  x x x x  x x x x    x x x x  x x x x
+              x x x x  x x x x    x x x x  x x x x  x x x x  x x x x    x x x x  x x x x
+              x x x x  x x x x    x x x x  x x x x  x x x x  x x x x    x x x x  x x x x
+              x x x x  x x x x    x x x x  x x x x  x x x x  x x x x    x x x x  x x x x
+              x x x x  x x x x    x x x x  x x x x  x x x x  x x x x    x x x x  x x x x
+              x x x x  x x x x    x x x x  x x x x  x x x x  x x x x    x x x x  x x x x
+              x x x x  x x x x    x x x x  x x x x  x x x x  x x x x    x x x x  x x x x
+              <0> <1>  <2> <3>    <4> <5>  <6> <7>  ..... WORK ITEMS
+
+              Each work item has 16 elements <8 rows and 2 cols of the original matrix>
+              the data_slice in holds the matrix elements in the following order:
+
+              0 0  0 0
+                /
+               /
+              1 1  1 1
+                /
+               /
+              2 2  2 2
+                /
+               / 
+              3 3  3 3 
+              
+              W0 --> 0 0 1 1 2 2 3 3 .... 7 7
+            */ 
+            // clang-format on
+
+           //  each WI calculates local sum of rows
+           // TM =8, TK = 32
+           for (int row = 0; row < TM * (TK / (SG_SZ * 2));
+                row++) { // there are 8 rows
+             for (int i = 0; i < data.length() / (SG_SZ / 2);
+                  i++) { // 2 per row
+               // i*SG_SIZE index is found based on the round robin
+               // distribution we are using in the implementation
+               global_index = row + global_idx * TM;
+               const auto data_index = row * 2 + i;
+               sum_local_rows[global_index] += data[data_index];
+               /* WI_global0 --> row 0 --> i [0,1] data [0, 1] --> local 0
+                  global 0 row 1 --> i [0,1] data [2, 3] --> local 1  global 1
+                          row 2 --> data [4, 5] --> local 2 global 2
+                          ..
+                          row 7 --> data [14, 15] --> local 7 global 7
+
+                  WI_global1 --> row 0 --> data [0, 1] --> local 8 global 8
+                          row 1 --> data [2, 3] --> local 9  global 9
+                          row 2 --> data [4, 5] --> local 10 global 10
+                          ..
+                          row 7 --> data [14, 15] --> local 15 global 15
+
+               */
+             }
+             sum_local_rows[global_index] = reduce_over_group(
+                 sg, sum_local_rows[global_index], sycl::plus<>());
+
+             // only Groups leader perform the global reduction
+             if (global_idy % SG_SZ == 0) {
+               atomic_fetch_add(v[global_index], sum_local_rows[global_index]);
+             }
+           }
+         }); // parallel for
```

Comment:
What exact functionality in the SYCL RT are we testing here that can't be covered with a more minimal test case?

Reply:
@aelovikov-intel aelovikov-intel, these tests have three goals:
- testing advanced element wise operations using joint matrix like sum of rows and sum of cols. The end goal is to add quantized GEMM that calls these functions.
- Testing the `get_coord` API here. While this has not been added yet in these examples, it will be added in the future once support is done in the backends. 
- provide a library style functions for users who want to use joint matrix for advanced computations like sum of rows

Reply:
I don't think this belongs here. llvml-test-suite should be directly targeting SYCL RT APIs and not include huge amounts of logic on top of that.

Reply:
@sergey-semenov , @bader what are your thoughts on this?